### ISSUE TITLE
NaN value is treated as a string in length frequency analysis

### ISSUE THREAD

**Issue by matejmojzes** [pandas-profiling/pandas-profiling#613]
It seems to me that a missing value (`NaN`) is treated as a string of 3 characters in the _Histogram of lengths of the category_ feature. I believe this behaviour is misleading, and thus would call it a bug.

**And example to be reproduced**

```
import numpy as np
import pandas as pd
from pandas_profiling import ProfileReport
df = pd.DataFrame([{'col': 'ABCD'}, {'col': 'DEFG'}, {'col': np.nan}])
ProfileReport(df)
```

Yields the following length statistics of the column `col`, which do not seem to be correct. 

Mean length = 3.666666667
Min length = 3

Or, graphically:

![image](https://user-images.githubusercontent.com/11783208/98408902-3b688c00-2072-11eb-8ed1-7724530d7ba4.png)

IMHO the empty value should not be taken into account in this kind of analysis.

**Version information**

* _Python version_: 3.8.3
* _Environment_: Local Jupyter Notebook
* pandas_profiling 2.9.0
* pandas 1.0.5
* numpy 1.18.5

**Comment by SunnyxBd** [comment]
Great catch @matejmojzes 
I created a PR that solves this bug https://github.com/pandas-profiling/pandas-profiling/pull/619

Cheers,
Sana

**Comment by sbrugman** [comment]
@matejmojzes I've just released v2.10.1 that should resolve your issue!

_Want me to continue to support this package? Please consider [sponsoring](https://github.com/sponsors/sbrugman)_
